block/manager.go

```
@@ -167,6 +172,30 @@ type Manager struct {
 	dataCommitmentToHeight sync.Map
 }
 
+// ManagerOption is a function that configures a Manager
+type ManagerOption func(*Manager)
+
+// WithValidatorHasher sets the validator hasher provider
+func WithValidatorHasher(hasher types.ValidatorHasher) ManagerOption {
+	return func(m *Manager) {
```

this and the other Options: prevent `nil` values -> fail fast